src/utils/codeSplitting.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * Dynamically imports a module and extracts a named export from it.
+ * @param {Promise} importPromise - A promise that resolves to the module.
+ * @param {string} namedExport -- The name of the export to extract
+ * @returns {Promise<any>} A promise that resolves to the named export.
+ */
+export default function extractNamedExport(importPromise, namedExport) {
```

Comment:
nit: if this is a default export, might recommend having the file name be `extract-named-export.js` or similar. Otherwise, perhaps consider it a named export so that other code-splitting helper constants/functions could also live in the same file as other named exports just like `extractNamedExport`.